pkg/kubelet/kubelet.go

```diff
@@ -3282,7 +3311,21 @@ func GetPhase(spec *api.PodSpec, info []api.ContainerStatus) api.PodPhase {
 
 func (kl *Kubelet) generateAPIPodStatus(pod *api.Pod, podStatus *kubecontainer.PodStatus) api.PodStatus {
 	glog.V(3).Infof("Generating status for %q", format.Pod(pod))
+
+	// any pod that requires eviction must be added to the list to sync.
+	attrs := &lifecycle.PodEvictAttributes{Pod: pod}
+	for _, podEvictor := range kl.podEvictors {
+		if result := podEvictor.Evict(attrs); result.Evict {
```

Comment:
> If we want to not wait for the housekeeping tick to ultimately force kill the pod, I think we can shrink that time by instead having the pod also potentially get killed during the syncPod function here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1743. I am fine with that if we want to check for a failed pod status in addition to canRunPod.

I think I suggested that in other comments as well. Somehow I missed this thread completely.

Another option is to explicitly pass this decision/information to the pod worker. Right now there is no way to instruct the workers to perform any operations other than *modifying the pod spec*. This may be better than relying on checking the "Failed" state in the long run.